packages/libs/sdk/src/solana/solana.ts

```diff
@@ -0,0 +1,146 @@
+// eslint-disable-next-line @nx/enforce-module-boundaries
+import {
+  Transaction,
+  Keypair,
+  ComputeBudgetProgram,
+  Connection,
+  TransactionInstruction,
+  PublicKey,
+  TransactionMessage,
+  VersionedTransaction,
+} from '@solana/web3.js';
+import {
+  type EstimatePriorityFeesParams,
+  type PriorityFeeRequestPayload,
+  type PriorityFeeResponseData,
+  type PriorityFeeLevels,
+  type SolanaClientArgs,
+} from './types';
+
+export class Solana {
+  readonly endpointUrl: string;
+  readonly connection: Connection;
+
+  constructor({ endpointUrl }: SolanaClientArgs) {
+    this.endpointUrl = endpointUrl;
+    this.connection = new Connection(endpointUrl);
+  }
+
+  async sendSmartTransaction(
+    transaction: Transaction,
+    keyPair: Keypair,
+    feeLevel: PriorityFeeLevels = 'medium'
+  ) {
+    const computeUnitPriceInstruction =
+      await this.createDynamicPriorityFeeInstruction(feeLevel);
+    const allInstructions = [
+      ...transaction.instructions,
+      computeUnitPriceInstruction,
+    ];
+
+    // eslint-disable-next-line prefer-const
+    let [units, recentBlockhash] = await Promise.all([
+      this.getSimulationUnits(
+        this.connection,
+        allInstructions,
+        keyPair.publicKey
+      ),
+      this.connection.getLatestBlockhash(),
+    ]);
+
+    transaction.add(computeUnitPriceInstruction);
+    if (units) {
+      units = Math.ceil(units * 1.05); // margin of error
+      transaction.add(ComputeBudgetProgram.setComputeUnitLimit({ units }));
+    }
```

Comment:
@johnpmitsch I'm wondering if insteading of (or maybe in addition) this being a `sendSmartTransaction` it's `assembleSmartTransaction` (or something like that), where we return the new `Transaction` object instead of sending it to the network. 
the benefit here is that many circumstances, you're dealing w/ a wallet, not a keypair. In that case, I think it might be more useful to be able to pass in my instructions (or transaction) and return the smart transaction...then I can sign/send with my wallet adapter. 
WDYT?